// @creativoma/liquid-glass/dist/index.esm.js
import ee, { useId as re, useState as te, useEffect as ne } from "react";
var h = { exports: {} }, _ = {};
var L;
function ae() {
  if (L) return _;
  L = 1;
  var i = Symbol.for("react.transitional.element"), c = Symbol.for("react.fragment");
  function u(d, s, l) {
    var f = null;
    if (l !== void 0 && (f = "" + l), s.key !== void 0 && (f = "" + s.key), "key" in s) {
      l = {};
      for (var m in s)
        m !== "key" && (l[m] = s[m]);
    } else l = s;
    return s = l.ref, {
      $$typeof: i,
      type: d,
      key: f,
      ref: s !== void 0 ? s : null,
      props: l
    };
  }
  return _.Fragment = c, _.jsx = u, _.jsxs = u, _;
}
var x = {};
var D;
function oe() {
  return D || (D = 1, process.env.NODE_ENV !== "production" && (function() {
    function i(e) {
      if (e == null) return null;
      if (typeof e == "function")
        return e.$$typeof === H ? null : e.displayName || e.name || null;
      if (typeof e == "string") return e;
      switch (e) {
        case y:
          return "Fragment";
        case W:
          return "Profiler";
        case U:
          return "StrictMode";
        case J:
          return "Suspense";
        case V:
          return "SuspenseList";
        case Z:
          return "Activity";
      }
      if (typeof e == "object")
        switch (typeof e.tag == "number" && console.error(
          "Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."
        ), e.$$typeof) {
          case G:
            return "Portal";
          case z:
            return e.displayName || "Context";
          case q:
            return (e._context.displayName || "Context") + ".Consumer";
          case B:
            var r = e.render;
            return e = e.displayName, e || (e = r.displayName || r.name || "", e = e !== "" ? "ForwardRef(" + e + ")" : "ForwardRef"), e;
          case X:
            return r = e.displayName || null, r !== null ? r : i(e.type) || "Memo";
          case k:
            r = e._payload, e = e._init;
            try {
              return i(e(r));
            } catch {
            }
        }
      return null;
    }
    function c(e) {
      return "" + e;
    }
    function u(e) {
      try {
        c(e);
        var r = !1;
      } catch {
        r = !0;
      }
      if (r) {
        r = console;
        var n = r.error, a = typeof Symbol == "function" && Symbol.toStringTag && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return n.call(
          r,
          "The provided key is an unsupported type %s. This value must be coerced to a string before using it here.",
          a
        ), c(e);
      }
    }
    function d(e) {
      if (e === y) return "<>";
      if (typeof e == "object" && e !== null && e.$$typeof === k)
        return "<...>";
      try {
        var r = i(e);
        return r ? "<" + r + ">" : "<...>";
      } catch {
        return "<...>";
      }
    }
    function s() {
      var e = S.A;
      return e === null ? null : e.getOwner();
    }
    function l() {
      return Error("react-stack-top-frame");
    }
    function f(e) {
      if (N.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning) return !1;
      }
      return e.key !== void 0;
    }
    function m(e, r) {
      function n() {
        C || (C = !0, console.error(
          "%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://react.dev/link/special-props)",
          r
        ));
      }
      n.isReactWarning = !0, Object.defineProperty(e, "key", {
        get: n,
        configurable: !0
      });
    }
    function T() {
      var e = i(this.type);
      return F[e] || (F[e] = !0, console.error(
        "Accessing element.ref was removed in React 19. ref is now a regular prop. It will be removed from the JSX Element type in a future release."
      )), e = this.props.ref, e !== void 0 ? e : null;
    }
    function g(e, r, n, a, R, O) {
      var o = n.ref;
      return e = {
        $$typeof: A,
        type: e,
        key: r,
        props: n,
        _owner: a
      }, (o !== void 0 ? o : null) !== null ? Object.defineProperty(e, "ref", {
        enumerable: !1,
        get: T
      }) : Object.defineProperty(e, "ref", { enumerable: !1, value: null }), e._store = {}, Object.defineProperty(e._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: 0
      }), Object.defineProperty(e, "_debugInfo", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: null
      }), Object.defineProperty(e, "_debugStack", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: R
      }), Object.defineProperty(e, "_debugTask", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: O
      }), Object.freeze && (Object.freeze(e.props), Object.freeze(e)), e;
    }
    function j(e, r, n, a, R, O) {
      var o = r.children;
      if (o !== void 0)
        if (a)
          if (Q(o)) {
            for (a = 0; a < o.length; a++)
              p(o[a]);
            Object.freeze && Object.freeze(o);
          } else
            console.error(
              "React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead."
            );
        else p(o);
      if (N.call(r, "key")) {
        o = i(e);
        var v = Object.keys(r).filter(function(K) {
          return K !== "key";
        });
        a = 0 < v.length ? "{key: someKey, " + v.join(": ..., ") + ": ...}" : "{key: someKey}", Y[o + a] || (v = 0 < v.length ? "{" + v.join(": ..., ") + ": ...}" : "{}", console.error(
          `A props object containing a "key" prop is being spread into JSX:
  let props = %s;
  <%s {...props} />
React keys must be passed directly to JSX without using spread:
  let props = %s;
  <%s key={someKey} {...props} />`,
          a,
          o,
          v,
          o
        ), Y[o + a] = !0);
      }
      if (o = null, n !== void 0 && (u(n), o = "" + n), f(r) && (u(r.key), o = "" + r.key), "key" in r) {
        n = {};
        for (var P in r)
          P !== "key" && (n[P] = r[P]);
      } else n = r;
      return o && m(
        n,
        typeof e == "function" ? e.displayName || e.name || "Unknown" : e
      ), g(
        e,
        o,
        n,
        s(),
        R,
        O
      );
    }
    function p(e) {
      E(e) ? e._store && (e._store.validated = 1) : typeof e == "object" && e !== null && e.$$typeof === k && (e._payload.status === "fulfilled" ? E(e._payload.value) && e._payload.value._store && (e._payload.value._store.validated = 1) : e._store && (e._store.validated = 1));
    }
    function E(e) {
      return typeof e == "object" && e !== null && e.$$typeof === A;
    }
    var b = ee, A = Symbol.for("react.transitional.element"), G = Symbol.for("react.portal"), y = Symbol.for("react.fragment"), U = Symbol.for("react.strict_mode"), W = Symbol.for("react.profiler"), q = Symbol.for("react.consumer"), z = Symbol.for("react.context"), B = Symbol.for("react.forward_ref"), J = Symbol.for("react.suspense"), V = Symbol.for("react.suspense_list"), X = Symbol.for("react.memo"), k = Symbol.for("react.lazy"), Z = Symbol.for("react.activity"), H = Symbol.for("react.client.reference"), S = b.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE, N = Object.prototype.hasOwnProperty, Q = Array.isArray, w = console.createTask ? console.createTask : function() {
      return null;
    };
    b = {
      react_stack_bottom_frame: function(e) {
        return e();
      }
    };
    var C, F = {}, $ = b.react_stack_bottom_frame.bind(
      b,
      l
    )(), I = w(d(l)), Y = {};
    x.Fragment = y, x.jsx = function(e, r, n) {
      var a = 1e4 > S.recentlyCreatedOwnerStacks++;
      return j(
        e,
        r,
        n,
        !1,
        a ? Error("react-stack-top-frame") : $,
        a ? w(d(e)) : I
      );
    }, x.jsxs = function(e, r, n) {
      var a = 1e4 > S.recentlyCreatedOwnerStacks++;
      return j(
        e,
        r,
        n,
        !0,
        a ? Error("react-stack-top-frame") : $,
        a ? w(d(e)) : I
      );
    };
  })()), x;
}
var M;
function se() {
  return M || (M = 1, process.env.NODE_ENV === "production" ? h.exports = ae() : h.exports = oe()), h.exports;
}
var t = se();
const le = () => {
  if (typeof window > "u") return !1;
  const i = window.navigator.userAgent, c = /iPad|iPhone|iPod/.test(i), u = /^((?!chrome|android).)*safari/i.test(i);
  return c || u;
}, ue = ({
  children: i,
  className: c = "",
  backdropBlur: u = 2,
  tintColor: d = "rgba(255, 255, 255, .2)",
  displacementScale: s = 150,
  turbulenceBaseFrequency: l = "0.008 0.008",
  turbulenceSeed: f = 1.5,
  as: m = "div",
  style: T,
  ...g
}) => {
  const p = `liquid-glass-${re().replace(/:/g, "-")}`, [E, b] = te(!1);
  return ne(() => {
    b(le());
  }, []), /* @__PURE__ */ t.jsxs(t.Fragment, { children: [
    /* @__PURE__ */ t.jsx("svg", { style: { display: "none" }, children: E ? (
      // Simplified filter for Safari/iOS - uses only well-supported primitives
      /* @__PURE__ */ t.jsxs(
        "filter",
        {
          id: p,
          x: "-20%",
          y: "-20%",
          width: "140%",
          height: "140%",
          filterUnits: "objectBoundingBox",
          children: [
            /* @__PURE__ */ t.jsx(
              "feTurbulence",
              {
                type: "fractalNoise",
                baseFrequency: l,
                numOctaves: 2,
                seed: f,
                result: "turbulence"
              }
            ),
            /* @__PURE__ */ t.jsx("feGaussianBlur", { in: "turbulence", stdDeviation: "2", result: "blur" }),
            /* @__PURE__ */ t.jsx(
              "feColorMatrix",
              {
                in: "blur",
                type: "matrix",
                values: `1 0 0 0 0
                      0 1 0 0 0
                      0 0 1 0 0
                      0 0 0 0.15 0`,
                result: "transparency"
              }
            ),
            /* @__PURE__ */ t.jsx("feBlend", { in: "SourceGraphic", in2: "transparency", mode: "normal" })
          ]
        }
      )
    ) : (
      // Full-featured filter for non-Safari browsers
      /* @__PURE__ */ t.jsxs(
        "filter",
        {
          id: p,
          x: "0%",
          y: "0%",
          width: "100%",
          height: "100%",
          filterUnits: "objectBoundingBox",
          children: [
            /* @__PURE__ */ t.jsx(
              "feTurbulence",
              {
                type: "fractalNoise",
                baseFrequency: l,
                numOctaves: 1,
                seed: f,
                result: "turbulence"
              }
            ),
            /* @__PURE__ */ t.jsxs("feComponentTransfer", { in: "turbulence", result: "mapped", children: [
              /* @__PURE__ */ t.jsx("feFuncR", { type: "gamma", amplitude: 1, exponent: 10, offset: 0.5 }),
              /* @__PURE__ */ t.jsx("feFuncG", { type: "gamma", amplitude: 0, exponent: 1, offset: 0 }),
              /* @__PURE__ */ t.jsx("feFuncB", { type: "gamma", amplitude: 0, exponent: 1, offset: 0.5 })
            ] }),
            /* @__PURE__ */ t.jsx("feGaussianBlur", { in: "turbulence", stdDeviation: 3, result: "softMap" }),
            /* @__PURE__ */ t.jsx(
              "feSpecularLighting",
              {
                in: "softMap",
                surfaceScale: 5,
                specularConstant: 1,
                specularExponent: 100,
                lightingColor: "white",
                result: "specLight",
                children: /* @__PURE__ */ t.jsx("fePointLight", { x: -200, y: -200, z: 300 })
              }
            ),
            /* @__PURE__ */ t.jsx(
              "feComposite",
              {
                in: "specLight",
                operator: "arithmetic",
                k1: 0,
                k2: 1,
                k3: 1,
                k4: 0,
                result: "litImage"
              }
            ),
            /* @__PURE__ */ t.jsx(
              "feDisplacementMap",
              {
                in: "SourceGraphic",
                in2: "softMap",
                scale: s,
                xChannelSelector: "R",
                yChannelSelector: "G"
              }
            )
          ]
        }
      )
    ) }),
    /* @__PURE__ */ t.jsxs(
      m,
      {
        className: `relative overflow-hidden ${c}`,
        style: {
          boxShadow: "0 6px 6px rgba(0, 0, 0, 0.2), 0 0 20px rgba(0, 0, 0, 0.1)",
          ...T
        },
        ...g,
        children: [
          /* @__PURE__ */ t.jsx(
            "div",
            {
              className: "absolute inset-0 z-0 overflow-hidden",
              style: {
                backdropFilter: `blur(${u}px)`,
                WebkitBackdropFilter: `blur(${u}px)`,
                filter: `url(#${p})`,
                isolation: "isolate",
                ...E && {
                  // Additional CSS-based effects for Safari/iOS to enhance the liquid glass appearance
                  transform: "translateZ(0)",
                  willChange: "transform"
                }
              }
            }
          ),
          /* @__PURE__ */ t.jsx(
            "div",
            {
              className: "absolute inset-0 z-[1]",
              style: {
                background: d
              }
            }
          ),
          /* @__PURE__ */ t.jsx("div", { className: "relative z-[2]", children: i })
        ]
      }
    )
  ] });
};
export {
  ue as LiquidGlass
};
