// .ignored/@mawtech/glass-ui/dist/index.mjs
import React4, { forwardRef, useState, useEffect, createContext, useRef, useCallback, useContext, useMemo } from 'react';
import { motion, AnimatePresence } from 'framer-motion';
import { cva } from 'class-variance-authority';
import { clsx } from 'clsx';
import { twMerge } from 'tailwind-merge';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import * as SelectPrimitive from '@radix-ui/react-select';
import * as CheckboxPrimitive from '@radix-ui/react-checkbox';
import * as SwitchPrimitive from '@radix-ui/react-switch';
import * as DialogPrimitive from '@radix-ui/react-dialog';
import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
import * as TooltipPrimitive from '@radix-ui/react-tooltip';
import * as TabsPrimitive from '@radix-ui/react-tabs';

// src/components/GlassCard/GlassCard.tsx
function cn(...inputs) {
  return twMerge(clsx(inputs));
}
var glassCardVariants = cva(
  // Base styles
  "relative overflow-hidden transition-all duration-250",
  {
    variants: {
      variant: {
        default: [
          "bg-glass-card/70 backdrop-blur-glass",
          "border border-white/12",
          "shadow-glass"
        ],
        elevated: [
          "bg-glass-card/80 backdrop-blur-glass-lg",
          "border border-white/15",
          "shadow-glass-lg"
        ],
        outlined: [
          "bg-glass-card/40 backdrop-blur-glass-sm",
          "border-2 border-white/20",
          "shadow-glass-sm"
        ],
        glow: [
          "bg-glass-card/70 backdrop-blur-glass",
          "border border-glass-cyan/30",
          "shadow-glass-glow"
        ],
        "glow-violet": [
          "bg-glass-card/70 backdrop-blur-glass",
          "border border-glass-violet/30",
          "shadow-glass-glow-violet"
        ],
        "glow-pink": [
          "bg-glass-card/70 backdrop-blur-glass",
          "border border-glass-pink/30",
          "shadow-glass-glow-pink"
        ]
      },
      padding: {
        none: "p-0",
        sm: "p-3",
        md: "p-5",
        lg: "p-7",
        xl: "p-9"
      },
      radius: {
        sm: "rounded-glass-sm",
        md: "rounded-xl",
        lg: "rounded-glass",
        xl: "rounded-glass-lg",
        full: "rounded-full"
      },
      hoverable: {
        true: "cursor-pointer hover:-translate-y-0.5 hover:shadow-glass-lg hover:border-white/20",
        false: ""
      }
    },
    defaultVariants: {
      variant: "default",
      padding: "md",
      radius: "lg",
      hoverable: false
    }
  }
);
var GlassCard = forwardRef(
  ({
    children,
    variant,
    padding,
    radius,
    hoverable,
    innerLight = true,
    disableAnimation = false,
    className,
    ...props
  }, ref) => {
    const animationProps = disableAnimation ? {} : {
      initial: { opacity: 0, y: 10 },
      animate: { opacity: 1, y: 0 },
      transition: { duration: 0.25, ease: [0.4, 0, 0.2, 1] }
    };
    return /* @__PURE__ */ jsx(
      motion.div,
      {
        ref,
        className: cn(
          glassCardVariants({ variant, padding, radius, hoverable }),
          innerLight && "shadow-[inset_0_1px_0_rgba(255,255,255,0.1)]",
          className
        ),
        ...animationProps,
        ...props,
        children
      }
    );
  }
);
GlassCard.displayName = "GlassCard";
var glassButtonVariants = cva(
  // Base styles
  [
    "relative inline-flex items-center justify-center gap-2",
    "font-medium transition-all duration-200",
    "focus:outline-none focus-visible:ring-2 focus-visible:ring-glass-cyan focus-visible:ring-offset-2 focus-visible:ring-offset-glass-dark",
    "disabled:opacity-50 disabled:cursor-not-allowed disabled:pointer-events-none"
  ],
  {
    variants: {
      variant: {
        primary: [
          "bg-glass-cyan text-glass-dark",
          "hover:bg-glass-cyan/90 hover:shadow-glass-glow",
          "active:scale-[0.98]"
        ],
        secondary: [
          "bg-glass-card/70 backdrop-blur-glass text-white",
          "border border-glass-cyan/50",
          "hover:bg-glass-card/90 hover:border-glass-cyan",
          "active:scale-[0.98]"
        ],
        ghost: [
          "bg-transparent text-white",
          "hover:bg-white/10",
          "active:bg-white/20"
        ],
        outline: [
          "bg-transparent text-white",
          "border border-white/20",
          "hover:bg-white/10 hover:border-white/40",
          "active:bg-white/20"
        ],
        danger: [
          "bg-red-500/80 backdrop-blur-glass text-white",
          "border border-red-400/50",
          "hover:bg-red-500 hover:shadow-[0_0_40px_rgba(239,68,68,0.3)]",
          "active:scale-[0.98]"
        ],
        success: [
          "bg-emerald-500/80 backdrop-blur-glass text-white",
          "border border-emerald-400/50",
          "hover:bg-emerald-500 hover:shadow-[0_0_40px_rgba(16,185,129,0.3)]",
          "active:scale-[0.98]"
        ],
        aurora: [
          "aurora-gradient text-white animate-aurora bg-[length:200%_200%]",
          "hover:shadow-glass-glow",
          "active:scale-[0.98]"
        ],
        violet: [
          "bg-glass-violet text-white",
          "hover:bg-glass-violet/90 hover:shadow-glass-glow-violet",
          "active:scale-[0.98]"
        ],
        pink: [
          "bg-glass-pink text-white",
          "hover:bg-glass-pink/90 hover:shadow-glass-glow-pink",
          "active:scale-[0.98]"
        ]
      },
      size: {
        xs: "h-7 px-2.5 text-xs rounded-lg",
        sm: "h-8 px-3 text-sm rounded-lg",
        md: "h-10 px-4 text-sm rounded-xl",
        lg: "h-12 px-6 text-base rounded-xl",
        xl: "h-14 px-8 text-lg rounded-2xl"
      },
      fullWidth: {
        true: "w-full",
        false: ""
      }
    },
    defaultVariants: {
      variant: "primary",
      size: "md",
      fullWidth: false
    }
  }
);
var GlassButton = forwardRef(
  ({
    children,
    variant,
    size,
    fullWidth,
    loading = false,
    leftIcon,
    rightIcon,
    disabled,
    disableHoverAnimation = false,
    className,
    ...props
  }, ref) => {
    const motionProps = disableHoverAnimation ? {} : {
      whileHover: { y: -2 },
      whileTap: { scale: 0.98 },
      transition: { duration: 0.15 }
    };
    return /* @__PURE__ */ jsx(
      motion.button,
      {
        ref,
        className: cn(
          glassButtonVariants({ variant, size, fullWidth }),
          className
        ),
        disabled: disabled || loading,
        ...motionProps,
        ...props,
        children: loading ? /* @__PURE__ */ jsxs(
          "svg",
          {
            className: "animate-spin h-4 w-4",
            xmlns: "http://www.w3.org/2000/svg",
            fill: "none",
            viewBox: "0 0 24 24",
            "aria-hidden": "true",
            children: [
              /* @__PURE__ */ jsx(
                "circle",
                {
                  className: "opacity-25",
                  cx: "12",
                  cy: "12",
                  r: "10",
                  stroke: "currentColor",
                  strokeWidth: "4"
                }
              ),
              /* @__PURE__ */ jsx(
                "path",
                {
                  className: "opacity-75",
                  fill: "currentColor",
                  d: "M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
                }
              )
            ]
          }
        ) : /* @__PURE__ */ jsxs(Fragment, { children: [
          leftIcon && /* @__PURE__ */ jsx("span", { className: "shrink-0", children: leftIcon }),
          children,
          rightIcon && /* @__PURE__ */ jsx("span", { className: "shrink-0", children: rightIcon })
        ] })
      }
    );
  }
);
GlassButton.displayName = "GlassButton";
var glassInputVariants = cva(
  [
    "w-full bg-glass-card/50 backdrop-blur-glass",
    "border border-white/12 rounded-xl",
    "text-white placeholder:text-white/40",
    "transition-all duration-200",
    "focus:outline-none focus:border-glass-cyan/50 focus:shadow-glass-glow",
    "disabled:opacity-50 disabled:cursor-not-allowed"
  ],
  {
    variants: {
      inputSize: {
        sm: "h-9 px-3 text-sm",
        md: "h-11 px-4 text-base",
        lg: "h-13 px-5 text-lg"
      },
      hasError: {
        true: "border-red-500/50 focus:border-red-500 focus:shadow-[0_0_40px_rgba(239,68,68,0.2)]",
        false: ""
      }
    },
    defaultVariants: {
      inputSize: "md",
      hasError: false
    }
  }
);
var GlassInput = forwardRef(
  ({
    label,
    helperText,
    error,
    inputSize,
    leftIcon,
    rightIcon,
    fullWidth = true,
    disabled,
    className,
    containerClassName,
    type = "text",
    id,
    ...props
  }, ref) => {
    const [showPassword, setShowPassword] = useState(false);
    const isPassword = type === "password";
    const inputType = isPassword ? showPassword ? "text" : "password" : type;
    const inputId = id || `glass-input-${Math.random().toString(36).substr(2, 9)}`;
    return /* @__PURE__ */ jsxs("div", { className: cn("flex flex-col gap-1.5", fullWidth && "w-full", containerClassName), children: [
      label && /* @__PURE__ */ jsx("label", { htmlFor: inputId, className: "text-sm font-medium text-white/80", children: label }),
      /* @__PURE__ */ jsxs("div", { className: "relative", children: [
        leftIcon && /* @__PURE__ */ jsx("div", { className: "absolute left-3 top-1/2 -translate-y-1/2 text-white/50", children: leftIcon }),
        /* @__PURE__ */ jsx(
          "input",
          {
            ref,
            id: inputId,
            type: inputType,
            disabled,
            "aria-invalid": !!error,
            "aria-describedby": error ? `${inputId}-error` : helperText ? `${inputId}-helper` : void 0,
            className: cn(
              glassInputVariants({ inputSize, hasError: !!error }),
              leftIcon && "pl-10",
              (rightIcon || isPassword) && "pr-10",
              className
            ),
            ...props
          }
        ),
        isPassword ? /* @__PURE__ */ jsx(
          "button",
          {
            type: "button",
            onClick: () => setShowPassword(!showPassword),
            className: "absolute right-3 top-1/2 -translate-y-1/2 text-white/50 hover:text-white transition-colors",
            "aria-label": showPassword ? "Hide password" : "Show password",
            children: showPassword ? /* @__PURE__ */ jsx("svg", { className: "w-5 h-5", fill: "none", stroke: "currentColor", viewBox: "0 0 24 24", children: /* @__PURE__ */ jsx(
              "path",
              {
                strokeLinecap: "round",
                strokeLinejoin: "round",
                strokeWidth: 2,
                d: "M13.875 18.825A10.05 10.05 0 0112 19c-4.478 0-8.268-2.943-9.543-7a9.97 9.97 0 011.563-3.029m5.858.908a3 3 0 114.243 4.243M9.878 9.878l4.242 4.242M9.88 9.88l-3.29-3.29m7.532 7.532l3.29 3.29M3 3l3.59 3.59m0 0A9.953 9.953 0 0112 5c4.478 0 8.268 2.943 9.543 7a10.025 10.025 0 01-4.132 5.411m0 0L21 21"
              }
            ) }) : /* @__PURE__ */ jsxs("svg", { className: "w-5 h-5", fill: "none", stroke: "currentColor", viewBox: "0 0 24 24", children: [
              /* @__PURE__ */ jsx(
                "path",
                {
                  strokeLinecap: "round",
                  strokeLinejoin: "round",
                  strokeWidth: 2,
                  d: "M15 12a3 3 0 11-6 0 3 3 0 016 0z"
                }
              ),
              /* @__PURE__ */ jsx(
                "path",
                {
                  strokeLinecap: "round",
                  strokeLinejoin: "round",
                  strokeWidth: 2,
                  d: "M2.458 12C3.732 7.943 7.523 5 12 5c4.478 0 8.268 2.943 9.542 7-1.274 4.057-5.064 7-9.542 7-4.477 0-8.268-2.943-9.542-7z"
                }
              )
            ] })
          }
        ) : rightIcon ? /* @__PURE__ */ jsx("div", { className: "absolute right-3 top-1/2 -translate-y-1/2 text-white/50", children: rightIcon }) : null
      ] }),
      (error || helperText) && /* @__PURE__ */ jsx(
        "p",
        {
          id: error ? `${inputId}-error` : `${inputId}-helper`,
          className: cn("text-sm", error ? "text-red-400" : "text-white/50"),
          role: error ? "alert" : void 0,
          children: error || helperText
        }
      )
    ] });
  }
);
GlassInput.displayName = "GlassInput";
var glassTextareaVariants = cva(
  [
    "w-full bg-glass-card/50 backdrop-blur-glass",
    "border border-white/12 rounded-xl",
    "text-white placeholder:text-white/40",
    "transition-all duration-200",
    "focus:outline-none focus:border-glass-cyan/50 focus:shadow-glass-glow",
    "disabled:opacity-50 disabled:cursor-not-allowed",
    "resize-none"
  ],
  {
    variants: {
      textareaSize: {
        sm: "p-3 text-sm min-h-[80px]",
        md: "p-4 text-base min-h-[120px]",
        lg: "p-5 text-lg min-h-[160px]"
      },
      hasError: {
        true: "border-red-500/50 focus:border-red-500 focus:shadow-[0_0_40px_rgba(239,68,68,0.2)]",
        false: ""
      },
      resize: {
        none: "resize-none",
        vertical: "resize-y",
        horizontal: "resize-x",
        both: "resize"
      }
    },
    defaultVariants: {
      textareaSize: "md",
      hasError: false,
      resize: "none"
    }
  }
);
var GlassTextarea = forwardRef(
  ({
    label,
    helperText,
    error,
    textareaSize,
    resize,
    showCount = false,
    maxLength,
    fullWidth = true,
    disabled,
    className,
    containerClassName,
    value,
    defaultValue,
    onChange,
    id,
    ...props
  }, ref) => {
    const [charCount, setCharCount] = useState(
      String(value || defaultValue || "").length
    );
    const textareaId = id || `glass-textarea-${Math.random().toString(36).substr(2, 9)}`;
    useEffect(() => {
      if (value !== void 0) {
        setCharCount(String(value).length);
      }
    }, [value]);
    const handleChange = (e) => {
      setCharCount(e.target.value.length);
      onChange?.(e);
    };
    return /* @__PURE__ */ jsxs("div", { className: cn("flex flex-col gap-1.5", fullWidth && "w-full", containerClassName), children: [
      label && /* @__PURE__ */ jsx("label", { htmlFor: textareaId, className: "text-sm font-medium text-white/80", children: label }),
      /* @__PURE__ */ jsx(
        "textarea",
        {
          ref,
          id: textareaId,
          disabled,
          maxLength,
          value,
          defaultValue,
          onChange: handleChange,
          "aria-invalid": !!error,
          "aria-describedby": error ? `${textareaId}-error` : helperText ? `${textareaId}-helper` : void 0,
          className: cn(
            glassTextareaVariants({ textareaSize, hasError: !!error, resize }),
            className
          ),
          ...props
        }
      ),
      /* @__PURE__ */ jsxs("div", { className: "flex justify-between items-center", children: [
        (error || helperText) && /* @__PURE__ */ jsx(
          "p",
          {
            id: error ? `${textareaId}-error` : `${textareaId}-helper`,
            className: cn("text-sm", error ? "text-red-400" : "text-white/50"),
            role: error ? "alert" : void 0,
            children: error || helperText
          }
        ),
        showCount && /* @__PURE__ */ jsxs(
          "p",
          {
            className: cn(
              "text-sm text-white/50 ml-auto",
              maxLength && charCount > maxLength && "text-red-400"
            ),
            children: [
              charCount,
              maxLength && `/${maxLength}`
            ]
          }
        )
      ] })
    ] });
  }
);
GlassTextarea.displayName = "GlassTextarea";
var glassSelectTriggerVariants = cva(
  [
    "flex items-center justify-between gap-2",
    "w-full bg-glass-card/50 backdrop-blur-glass",
    "border border-white/12 rounded-xl",
    "text-white",
    "transition-all duration-200",
    "focus:outline-none focus:border-glass-cyan/50 focus:shadow-glass-glow",
    "disabled:opacity-50 disabled:cursor-not-allowed",
    "data-[placeholder]:text-white/40"
  ],
  {
    variants: {
      selectSize: {
        sm: "h-9 px-3 text-sm",
        md: "h-11 px-4 text-base",
        lg: "h-13 px-5 text-lg"
      },
      hasError: {
        true: "border-red-500/50 focus:border-red-500 focus:shadow-[0_0_40px_rgba(239,68,68,0.2)]",
        false: ""
      }
    },
    defaultVariants: {
      selectSize: "md",
      hasError: false
    }
  }
);
var SelectIcon = () => /* @__PURE__ */ jsx(
  "svg",
  {
    className: "w-4 h-4 text-white/50",
    fill: "none",
    stroke: "currentColor",
    viewBox: "0 0 24 24",
    children: /* @__PURE__ */ jsx(
      "path",
      {
        strokeLinecap: "round",
        strokeLinejoin: "round",
        strokeWidth: 2,
        d: "M19 9l-7 7-7-7"
      }
    )
  }
);
var CheckIcon = () => /* @__PURE__ */ jsx(
  "svg",
  {
    className: "w-4 h-4",
    fill: "none",
    stroke: "currentColor",
    viewBox: "0 0 24 24",
    children: /* @__PURE__ */ jsx(
      "path",
      {
        strokeLinecap: "round",
        strokeLinejoin: "round",
        strokeWidth: 2,
        d: "M5 13l4 4L19 7"
      }
    )
  }
);
var GlassSelect = forwardRef(
  ({
    label,
    placeholder = "Select an option",
    options,
    helperText,
    error,
    selectSize,
    fullWidth = true,
    containerClassName,
    triggerClassName,
    disabled,
    ...props
  }, ref) => {
    const selectId = `glass-select-${Math.random().toString(36).substr(2, 9)}`;
    return /* @__PURE__ */ jsxs(
      "div",
      {
        className: cn(
          "flex flex-col gap-1.5",
          fullWidth && "w-full",
          containerClassName
        ),
        children: [
          label && /* @__PURE__ */ jsx(
            "label",
            {
              htmlFor: selectId,
              className: "text-sm font-medium text-white/80",
              children: label
            }
          ),
          /* @__PURE__ */ jsxs(SelectPrimitive.Root, { disabled, ...props, children: [
            /* @__PURE__ */ jsxs(
              SelectPrimitive.Trigger,
              {
                ref,
                id: selectId,
                className: cn(
                  glassSelectTriggerVariants({ selectSize, hasError: !!error }),
                  triggerClassName
                ),
                "aria-invalid": !!error,
                "aria-describedby": error ? `${selectId}-error` : helperText ? `${selectId}-helper` : void 0,
                children: [
                  /* @__PURE__ */ jsx(SelectPrimitive.Value, { placeholder }),
                  /* @__PURE__ */ jsx(SelectPrimitive.Icon, { asChild: true, children: /* @__PURE__ */ jsx(SelectIcon, {}) })
                ]
              }
            ),
            /* @__PURE__ */ jsx(SelectPrimitive.Portal, { children: /* @__PURE__ */ jsx(
              SelectPrimitive.Content,
              {
                className: cn(
                  "overflow-hidden rounded-xl",
                  "bg-glass-card/95 backdrop-blur-glass-lg",
                  "border border-white/12",
                  "shadow-glass-lg",
                  "animate-scale-in",
                  "z-50"
                ),
                position: "popper",
                sideOffset: 4,
                children: /* @__PURE__ */ jsx(SelectPrimitive.Viewport, { className: "p-1", children: options.map((option) => /* @__PURE__ */ jsxs(
                  SelectPrimitive.Item,
                  {
                    value: option.value,
                    disabled: option.disabled,
                    className: cn(
                      "relative flex items-center px-3 py-2 rounded-lg",
                      "text-sm text-white",
                      "cursor-pointer select-none outline-none",
                      "data-[highlighted]:bg-white/10",
                      "data-[disabled]:opacity-50 data-[disabled]:cursor-not-allowed",
                      "transition-colors duration-150"
                    ),
                    children: [
                      /* @__PURE__ */ jsx(SelectPrimitive.ItemText, { children: option.label }),
                      /* @__PURE__ */ jsx(SelectPrimitive.ItemIndicator, { className: "absolute right-2", children: /* @__PURE__ */ jsx(CheckIcon, {}) })
                    ]
                  },
                  option.value
                )) })
              }
            ) })
          ] }),
          (error || helperText) && /* @__PURE__ */ jsx(
            "p",
            {
              id: error ? `${selectId}-error` : `${selectId}-helper`,
              className: cn("text-sm", error ? "text-red-400" : "text-white/50"),
              role: error ? "alert" : void 0,
              children: error || helperText
            }
          )
        ]
      }
    );
  }
);
GlassSelect.displayName = "GlassSelect";
var glassCheckboxVariants = cva(
  [
    "flex items-center justify-center",
    "bg-glass-card/50 backdrop-blur-glass",
    "border border-white/20 rounded-md",
    "transition-all duration-200",
    "focus:outline-none focus-visible:ring-2 focus-visible:ring-glass-cyan focus-visible:ring-offset-2 focus-visible:ring-offset-glass-dark",
    "disabled:opacity-50 disabled:cursor-not-allowed",
    "data-[state=checked]:bg-glass-cyan data-[state=checked]:border-glass-cyan",
    "hover:border-white/40",
    "data-[state=checked]:hover:bg-glass-cyan/90"
  ],
  {
    variants: {
      checkboxSize: {
        sm: "w-4 h-4",
        md: "w-5 h-5",
        lg: "w-6 h-6"
      }
    },
    defaultVariants: {
      checkboxSize: "md"
    }
  }
);
var CheckIcon2 = ({ size }) => {
  const sizes = {
    sm: "w-2.5 h-2.5",
    md: "w-3 h-3",
    lg: "w-4 h-4"
  };
  return /* @__PURE__ */ jsx(
    motion.svg,
    {
      className: cn(sizes[size], "text-glass-dark"),
      fill: "none",
      stroke: "currentColor",
      viewBox: "0 0 24 24",
      initial: { scale: 0, opacity: 0 },
      animate: { scale: 1, opacity: 1 },
      exit: { scale: 0, opacity: 0 },
      transition: { duration: 0.15, ease: [0.4, 0, 0.2, 1] },
      children: /* @__PURE__ */ jsx(
        motion.path,
        {
          strokeLinecap: "round",
          strokeLinejoin: "round",
          strokeWidth: 3,
          d: "M5 13l4 4L19 7",
          initial: { pathLength: 0 },
          animate: { pathLength: 1 },
          transition: { duration: 0.2, delay: 0.1 }
        }
      )
    }
  );
};
var GlassCheckbox = forwardRef(
  ({
    label,
    description,
    checkboxSize = "md",
    disabled,
    className,
    containerClassName,
    id,
    ...props
  }, ref) => {
    const checkboxId = id || `glass-checkbox-${Math.random().toString(36).substr(2, 9)}`;
    return /* @__PURE__ */ jsxs("div", { className: cn("flex items-start gap-3", containerClassName), children: [
      /* @__PURE__ */ jsx(
        CheckboxPrimitive.Root,
        {
          ref,
          id: checkboxId,
          disabled,
          className: cn(glassCheckboxVariants({ checkboxSize }), className),
          ...props,
          children: /* @__PURE__ */ jsx(CheckboxPrimitive.Indicator, { asChild: true, children: /* @__PURE__ */ jsx(AnimatePresence, { mode: "wait", children: /* @__PURE__ */ jsx(CheckIcon2, { size: checkboxSize || "md" }) }) })
        }
      ),
      (label || description) && /* @__PURE__ */ jsxs("div", { className: "flex flex-col gap-0.5", children: [
        label && /* @__PURE__ */ jsx(
          "label",
          {
            htmlFor: checkboxId,
            className: cn(
              "text-sm font-medium text-white cursor-pointer",
              disabled && "opacity-50 cursor-not-allowed"
            ),
            children: label
          }
        ),
        description && /* @__PURE__ */ jsx(
          "p",
          {
            className: cn(
              "text-sm text-white/50",
              disabled && "opacity-50"
            ),
            children: description
          }
        )
      ] })
    ] });
  }
);
GlassCheckbox.displayName = "GlassCheckbox";
var glassSwitchVariants = cva(
  [
    "relative inline-flex shrink-0 cursor-pointer",
    "bg-glass-card/70 backdrop-blur-glass",
    "border border-white/20 rounded-full",
    "transition-all duration-200",
    "focus:outline-none focus-visible:ring-2 focus-visible:ring-glass-cyan focus-visible:ring-offset-2 focus-visible:ring-offset-glass-dark",
    "disabled:opacity-50 disabled:cursor-not-allowed",
    "data-[state=checked]:bg-glass-cyan data-[state=checked]:border-glass-cyan"
  ],
  {
    variants: {
      switchSize: {
        sm: "w-8 h-5",
        md: "w-11 h-6",
        lg: "w-14 h-8"
      }
    },
    defaultVariants: {
      switchSize: "md"
    }
  }
);
var glassSwitchThumbVariants = cva(
  [
    "block rounded-full bg-white shadow-lg",
    "transition-transform duration-200",
    "data-[state=checked]:translate-x-full"
  ],
  {
    variants: {
      switchSize: {
        sm: "w-3.5 h-3.5 data-[state=unchecked]:translate-x-0.5 data-[state=checked]:translate-x-[calc(100%-2px)]",
        md: "w-5 h-5 data-[state=unchecked]:translate-x-0.5 data-[state=checked]:translate-x-[calc(100%-2px)]",
        lg: "w-6 h-6 data-[state=unchecked]:translate-x-1 data-[state=checked]:translate-x-[calc(100%)]"
      }
    },
    defaultVariants: {
      switchSize: "md"
    }
  }
);
var GlassSwitch = forwardRef(
  ({
    label,
    description,
    switchSize = "md",
    labelPosition = "right",
    disabled,
    className,
    containerClassName,
    id,
    ...props
  }, ref) => {
    const switchId = id || `glass-switch-${Math.random().toString(36).substr(2, 9)}`;
    const LabelContent = () => (label || description) && /* @__PURE__ */ jsxs("div", { className: "flex flex-col gap-0.5", children: [
      label && /* @__PURE__ */ jsx(
        "label",
        {
          htmlFor: switchId,
          className: cn(
            "text-sm font-medium text-white cursor-pointer",
            disabled && "opacity-50 cursor-not-allowed"
          ),
          children: label
        }
      ),
      description && /* @__PURE__ */ jsx(
        "p",
        {
          className: cn("text-sm text-white/50", disabled && "opacity-50"),
          children: description
        }
      )
    ] });
    return /* @__PURE__ */ jsxs(
      "div",
      {
        className: cn(
          "flex items-center gap-3",
          labelPosition === "left" && "flex-row-reverse justify-end",
          containerClassName
        ),
        children: [
          /* @__PURE__ */ jsx(
            SwitchPrimitive.Root,
            {
              ref,
              id: switchId,
              disabled,
              className: cn(glassSwitchVariants({ switchSize }), className),
              ...props,
              children: /* @__PURE__ */ jsx(
                SwitchPrimitive.Thumb,
                {
                  className: glassSwitchThumbVariants({ switchSize })
                }
              )
            }
          ),
          /* @__PURE__ */ jsx(LabelContent, {})
        ]
      }
    );
  }
);
GlassSwitch.displayName = "GlassSwitch";
var glassModalVariants = cva(
  [
    "fixed left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2",
    "w-full bg-glass-card/90 backdrop-blur-glass-lg",
    "border border-white/12 rounded-glass",
    "shadow-glass-lg",
    "p-6",
    "z-50",
    "focus:outline-none"
  ],
  {
    variants: {
      size: {
        sm: "max-w-sm",
        md: "max-w-md",
        lg: "max-w-lg",
        xl: "max-w-xl",
        "2xl": "max-w-2xl",
        full: "max-w-[calc(100%-2rem)] max-h-[calc(100%-2rem)]"
      }
    },
    defaultVariants: {
      size: "md"
    }
  }
);
var CloseIcon = () => /* @__PURE__ */ jsx(
  "svg",
  {
    className: "w-4 h-4",
    fill: "none",
    stroke: "currentColor",
    viewBox: "0 0 24 24",
    children: /* @__PURE__ */ jsx(
      "path",
      {
        strokeLinecap: "round",
        strokeLinejoin: "round",
        strokeWidth: 2,
        d: "M6 18L18 6M6 6l12 12"
      }
    )
  }
);
var overlayVariants = {
  hidden: { opacity: 0 },
  visible: { opacity: 1 }
};
var contentVariants = {
  hidden: { opacity: 0, scale: 0.95, y: -10 },
  visible: { opacity: 1, scale: 1, y: 0 }
};
var GlassModal = forwardRef(
  ({
    title,
    description,
    children,
    size,
    showClose = true,
    contentClassName,
    trigger,
    open,
    onOpenChange,
    ...props
  }, ref) => {
    return /* @__PURE__ */ jsxs(DialogPrimitive.Root, { open, onOpenChange, ...props, children: [
      trigger && /* @__PURE__ */ jsx(DialogPrimitive.Trigger, { asChild: true, children: trigger }),
      /* @__PURE__ */ jsx(AnimatePresence, { children: open !== false && /* @__PURE__ */ jsxs(DialogPrimitive.Portal, { forceMount: true, children: [
        /* @__PURE__ */ jsx(DialogPrimitive.Overlay, { asChild: true, children: /* @__PURE__ */ jsx(
          motion.div,
          {
            className: "fixed inset-0 bg-black/60 backdrop-blur-sm z-50",
            variants: overlayVariants,
            initial: "hidden",
            animate: "visible",
            exit: "hidden",
            transition: { duration: 0.2 }
          }
        ) }),
        /* @__PURE__ */ jsx(DialogPrimitive.Content, { asChild: true, children: /* @__PURE__ */ jsxs(
          motion.div,
          {
            ref,
            className: cn(glassModalVariants({ size }), contentClassName),
            variants: contentVariants,
            initial: "hidden",
            animate: "visible",
            exit: "hidden",
            transition: { duration: 0.2, ease: [0.4, 0, 0.2, 1] },
            children: [
              showClose && /* @__PURE__ */ jsx(
                DialogPrimitive.Close,
                {
                  className: cn(
                    "absolute top-4 right-4",
                    "p-2 rounded-lg",
                    "text-white/50 hover:text-white hover:bg-white/10",
                    "transition-colors duration-150",
                    "focus:outline-none focus-visible:ring-2 focus-visible:ring-glass-cyan"
                  ),
                  "aria-label": "Close",
                  children: /* @__PURE__ */ jsx(CloseIcon, {})
                }
              ),
              title && /* @__PURE__ */ jsx(DialogPrimitive.Title, { className: "text-xl font-semibold text-white mb-2", children: title }),
              description && /* @__PURE__ */ jsx(DialogPrimitive.Description, { className: "text-sm text-white/60 mb-4", children: description }),
              children
            ]
          }
        ) })
      ] }) })
    ] });
  }
);
GlassModal.displayName = "GlassModal";
var GlassModalTrigger = DialogPrimitive.Trigger;
var GlassModalClose = DialogPrimitive.Close;
var glassNavbarVariants = cva(
  [
    "w-full bg-glass-card/70 backdrop-blur-glass",
    "border-b border-white/12",
    "shadow-glass-sm"
  ],
  {
    variants: {
      position: {
        static: "relative",
        fixed: "fixed top-0 left-0 right-0 z-50",
        sticky: "sticky top-0 z-50"
      },
      size: {
        sm: "h-14",
        md: "h-16",
        lg: "h-20"
      }
    },
    defaultVariants: {
      position: "static",
      size: "md"
    }
  }
);
var glassNavItemVariants = cva(
  [
    "relative px-4 py-2 rounded-lg",
    "text-sm font-medium text-white/70",
    "transition-all duration-200",
    "hover:text-white hover:bg-white/10",
    "focus:outline-none focus-visible:ring-2 focus-visible:ring-glass-cyan"
  ],
  {
    variants: {
      active: {
        true: "text-white bg-white/10",
        false: ""
      }
    },
    defaultVariants: {
      active: false
    }
  }
);
var GlassNavItem = forwardRef(
  ({ children, active, icon, href, className, ...props }, ref) => {
    const content = /* @__PURE__ */ jsxs(Fragment, { children: [
      icon && /* @__PURE__ */ jsx("span", { className: "mr-2", children: icon }),
      children,
      active && /* @__PURE__ */ jsx(
        motion.div,
        {
          className: "absolute bottom-0 left-2 right-2 h-0.5 bg-glass-cyan rounded-full",
          layoutId: "navbar-indicator",
          transition: { duration: 0.2 }
        }
      )
    ] });
    if (href) {
      return /* @__PURE__ */ jsx(
        "a",
        {
          href,
          className: cn(glassNavItemVariants({ active }), className),
          children: content
        }
      );
    }
    return /* @__PURE__ */ jsx(
      "button",
      {
        ref,
        className: cn(glassNavItemVariants({ active }), className),
        ...props,
        children: content
      }
    );
  }
);
GlassNavItem.displayName = "GlassNavItem";
var MenuIcon = () => /* @__PURE__ */ jsx("svg", { className: "w-6 h-6", fill: "none", stroke: "currentColor", viewBox: "0 0 24 24", children: /* @__PURE__ */ jsx("path", { strokeLinecap: "round", strokeLinejoin: "round", strokeWidth: 2, d: "M4 6h16M4 12h16M4 18h16" }) });
var CloseIcon2 = () => /* @__PURE__ */ jsx("svg", { className: "w-6 h-6", fill: "none", stroke: "currentColor", viewBox: "0 0 24 24", children: /* @__PURE__ */ jsx("path", { strokeLinecap: "round", strokeLinejoin: "round", strokeWidth: 2, d: "M6 18L18 6M6 6l12 12" }) });
var GlassNavbar = forwardRef(
  ({
    logo,
    children,
    actions,
    position,
    size,
    maxWidth = "xl",
    className,
    ...props
  }, ref) => {
    const [mobileMenuOpen, setMobileMenuOpen] = useState(false);
    const maxWidthClasses = {
      sm: "max-w-screen-sm",
      md: "max-w-screen-md",
      lg: "max-w-screen-lg",
      xl: "max-w-screen-xl",
      "2xl": "max-w-screen-2xl",
      full: "max-w-full"
    };
    return /* @__PURE__ */ jsxs(
      "nav",
      {
        ref,
        className: cn(glassNavbarVariants({ position, size }), className),
        ...props,
        children: [
          /* @__PURE__ */ jsxs(
            "div",
            {
              className: cn(
                "flex items-center justify-between h-full px-4 mx-auto",
                maxWidthClasses[maxWidth]
              ),
              children: [
                logo && /* @__PURE__ */ jsx("div", { className: "shrink-0", children: logo }),
                /* @__PURE__ */ jsx("div", { className: "hidden md:flex items-center gap-1 flex-1 justify-center", children }),
                /* @__PURE__ */ jsx("div", { className: "hidden md:flex items-center gap-4", children: actions }),
                /* @__PURE__ */ jsx(
                  "button",
                  {
                    className: "md:hidden p-2 text-white/70 hover:text-white hover:bg-white/10 rounded-lg transition-colors",
                    onClick: () => setMobileMenuOpen(!mobileMenuOpen),
                    "aria-label": mobileMenuOpen ? "Close menu" : "Open menu",
                    "aria-expanded": mobileMenuOpen,
                    children: mobileMenuOpen ? /* @__PURE__ */ jsx(CloseIcon2, {}) : /* @__PURE__ */ jsx(MenuIcon, {})
                  }
                )
              ]
            }
          ),
          /* @__PURE__ */ jsx(AnimatePresence, { children: mobileMenuOpen && /* @__PURE__ */ jsx(
            motion.div,
            {
              initial: { opacity: 0, height: 0 },
              animate: { opacity: 1, height: "auto" },
              exit: { opacity: 0, height: 0 },
              transition: { duration: 0.2 },
              className: "md:hidden overflow-hidden bg-glass-card/90 backdrop-blur-glass border-t border-white/12",
              children: /* @__PURE__ */ jsxs("div", { className: "flex flex-col p-4 gap-2", children: [
                children,
                actions && /* @__PURE__ */ jsx("div", { className: "pt-4 border-t border-white/12 mt-2", children: actions })
              ] })
            }
          ) })
        ]
      }
    );
  }
);
GlassNavbar.displayName = "GlassNavbar";
var glassSidebarVariants = cva(
  [
    "flex flex-col",
    "bg-glass-card/70 backdrop-blur-glass",
    "border-r border-white/12",
    "shadow-glass",
    "transition-all duration-300"
  ],
  {
    variants: {
      position: {
        left: "left-0",
        right: "right-0 border-r-0 border-l"
      },
      variant: {
        default: "",
        floating: "rounded-glass m-4 border shadow-glass-lg"
      }
    },
    defaultVariants: {
      position: "left",
      variant: "default"
    }
  }
);
var glassSidebarItemVariants = cva(
  [
    "relative flex items-center gap-3 px-4 py-3 rounded-xl",
    "text-sm font-medium text-white/70",
    "transition-all duration-200",
    "hover:text-white hover:bg-white/10",
    "focus:outline-none focus-visible:ring-2 focus-visible:ring-glass-cyan focus-visible:ring-inset"
  ],
  {
    variants: {
      active: {
        true: "text-white bg-white/10",
        false: ""
      }
    },
    defaultVariants: {
      active: false
    }
  }
);
var SidebarContext = createContext(
  void 0
);
var useSidebar = () => {
  const context = useContext(SidebarContext);
  if (!context) {
    throw new Error("Sidebar components must be used within GlassSidebar");
  }
  return context;
};
var GlassSidebarItem = forwardRef(({ children, active, icon, href, badge, className, ...props }, ref) => {
  const { collapsed } = useSidebar();
  const content = /* @__PURE__ */ jsxs(Fragment, { children: [
    icon && /* @__PURE__ */ jsx("span", { className: "shrink-0 w-5 h-5", children: icon }),
    /* @__PURE__ */ jsx(AnimatePresence, { mode: "wait", children: !collapsed && /* @__PURE__ */ jsx(
      motion.span,
      {
        initial: { opacity: 0, width: 0 },
        animate: { opacity: 1, width: "auto" },
        exit: { opacity: 0, width: 0 },
        transition: { duration: 0.2 },
        className: "truncate",
        children
      }
    ) }),
    badge && !collapsed && /* @__PURE__ */ jsx("span", { className: "ml-auto shrink-0", children: badge }),
    active && /* @__PURE__ */ jsx(
      motion.div,
      {
        className: "absolute left-0 top-1/2 -translate-y-1/2 w-1 h-6 bg-glass-cyan rounded-r-full",
        layoutId: "sidebar-indicator",
        transition: { duration: 0.2 }
      }
    )
  ] });
  const itemClasses = cn(
    glassSidebarItemVariants({ active }),
    collapsed && "justify-center px-3",
    className
  );
  if (href) {
    return /* @__PURE__ */ jsx("a", { href, className: itemClasses, children: content });
  }
  return /* @__PURE__ */ jsx("button", { ref, className: itemClasses, ...props, children: content });
});
GlassSidebarItem.displayName = "GlassSidebarItem";
var ChevronIcon = ({ direction }) => /* @__PURE__ */ jsx(
  "svg",
  {
    className: cn("w-4 h-4", direction === "left" && "rotate-180"),
    fill: "none",
    stroke: "currentColor",
    viewBox: "0 0 24 24",
    children: /* @__PURE__ */ jsx(
      "path",
      {
        strokeLinecap: "round",
        strokeLinejoin: "round",
        strokeWidth: 2,
        d: "M9 5l7 7-7 7"
      }
    )
  }
);
var GlassSidebar = forwardRef(
  ({
    header,
    footer,
    children,
    position,
    variant,
    collapsed: controlledCollapsed,
    onCollapsedChange,
    defaultCollapsed = false,
    collapsible = false,
    width = 260,
    collapsedWidth = 72,
    className,
    ...props
  }, ref) => {
    const [internalCollapsed, setInternalCollapsed] = useState(defaultCollapsed);
    const collapsed = controlledCollapsed ?? internalCollapsed;
    const setCollapsed = (value) => {
      setInternalCollapsed(value);
      onCollapsedChange?.(value);
    };
    const currentWidth = collapsed ? collapsedWidth : width;
    return /* @__PURE__ */ jsx(SidebarContext.Provider, { value: { collapsed, setCollapsed }, children: /* @__PURE__ */ jsxs(
      motion.aside,
      {
        ref,
        animate: { width: currentWidth },
        transition: { duration: 0.3, ease: [0.4, 0, 0.2, 1] },
        className: cn(
          glassSidebarVariants({ position, variant }),
          "h-full overflow-hidden",
          className
        ),
        ...props,
        children: [
          header && /* @__PURE__ */ jsx(
            "div",
            {
              className: cn(
                "p-4 border-b border-white/12",
                collapsed && "px-3"
              ),
              children: header
            }
          ),
          /* @__PURE__ */ jsx("nav", { className: "flex-1 p-3 space-y-1 overflow-y-auto", children }),
          collapsible && /* @__PURE__ */ jsx("div", { className: "p-3 border-t border-white/12", children: /* @__PURE__ */ jsxs(
            "button",
            {
              onClick: () => setCollapsed(!collapsed),
              className: cn(
                "w-full flex items-center gap-3 px-4 py-3 rounded-xl",
                "text-sm font-medium text-white/70",
                "hover:text-white hover:bg-white/10",
                "transition-all duration-200",
                collapsed && "justify-center px-3"
              ),
              "aria-label": collapsed ? "Expand sidebar" : "Collapse sidebar",
              children: [
                /* @__PURE__ */ jsx(ChevronIcon, { direction: collapsed ? "right" : "left" }),
                /* @__PURE__ */ jsx(AnimatePresence, { mode: "wait", children: !collapsed && /* @__PURE__ */ jsx(
                  motion.span,
                  {
                    initial: { opacity: 0, width: 0 },
                    animate: { opacity: 1, width: "auto" },
                    exit: { opacity: 0, width: 0 },
                    transition: { duration: 0.2 },
                    children: "Collapse"
                  }
                ) })
              ]
            }
          ) }),
          footer && /* @__PURE__ */ jsx(
            "div",
            {
              className: cn(
                "p-4 border-t border-white/12",
                collapsed && "px-3"
              ),
              children: footer
            }
          )
        ]
      }
    ) });
  }
);
GlassSidebar.displayName = "GlassSidebar";
var contentVariants2 = {
  hidden: { opacity: 0, scale: 0.95 },
  visible: { opacity: 1, scale: 1 }
};
var GlassDropdownItem = forwardRef(({ children, icon, shortcut, destructive, className, ...props }, ref) => {
  return /* @__PURE__ */ jsxs(
    DropdownMenuPrimitive.Item,
    {
      ref,
      className: cn(
        "relative flex items-center gap-2 px-3 py-2 rounded-lg",
        "text-sm text-white outline-none cursor-pointer",
        "data-[highlighted]:bg-white/10",
        "data-[disabled]:opacity-50 data-[disabled]:cursor-not-allowed",
        "transition-colors duration-150",
        destructive && "text-red-400 data-[highlighted]:bg-red-500/20",
        className
      ),
      ...props,
      children: [
        icon && /* @__PURE__ */ jsx("span", { className: "w-4 h-4 shrink-0", children: icon }),
        /* @__PURE__ */ jsx("span", { className: "flex-1", children }),
        shortcut && /* @__PURE__ */ jsx("span", { className: "ml-auto text-xs text-white/40", children: shortcut })
      ]
    }
  );
});
GlassDropdownItem.displayName = "GlassDropdownItem";
var GlassDropdownSeparator = forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx(
  DropdownMenuPrimitive.Separator,
  {
    ref,
    className: cn("h-px my-1 bg-white/10", className),
    ...props
  }
));
GlassDropdownSeparator.displayName = "GlassDropdownSeparator";
var GlassDropdownLabel = forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx(
  DropdownMenuPrimitive.Label,
  {
    ref,
    className: cn("px-3 py-2 text-xs font-medium text-white/50", className),
    ...props
  }
));
GlassDropdownLabel.displayName = "GlassDropdownLabel";
var GlassDropdown = forwardRef(
  ({
    trigger,
    children,
    align = "start",
    side = "bottom",
    sideOffset = 4,
    open,
    onOpenChange,
    ...props
  }, ref) => {
    return /* @__PURE__ */ jsxs(
      DropdownMenuPrimitive.Root,
      {
        open,
        onOpenChange,
        ...props,
        children: [
          /* @__PURE__ */ jsx(DropdownMenuPrimitive.Trigger, { asChild: true, children: trigger }),
          /* @__PURE__ */ jsx(AnimatePresence, { children: /* @__PURE__ */ jsx(DropdownMenuPrimitive.Portal, { children: /* @__PURE__ */ jsx(
            DropdownMenuPrimitive.Content,
            {
              ref,
              align,
              side,
              sideOffset,
              asChild: true,
              children: /* @__PURE__ */ jsx(
                motion.div,
                {
                  className: cn(
                    "min-w-[180px] p-1 rounded-xl",
                    "bg-glass-card/95 backdrop-blur-glass-lg",
                    "border border-white/12",
                    "shadow-glass-lg",
                    "z-50"
                  ),
                  variants: contentVariants2,
                  initial: "hidden",
                  animate: "visible",
                  exit: "hidden",
                  transition: { duration: 0.15, ease: [0.4, 0, 0.2, 1] },
                  children
                }
              )
            }
          ) }) })
        ]
      }
    );
  }
);
GlassDropdown.displayName = "GlassDropdown";
var glassTooltipVariants = cva(
  [
    "px-3 py-2 rounded-lg",
    "bg-glass-card backdrop-blur-glass",
    "shadow-[0_4px_12px_rgba(0,0,0,0.4),inset_0_0_0_1px_rgba(255,255,255,0.1)]",
    "text-sm text-white",
    "z-50",
    "animate-scale-in"
  ],
  {
    variants: {
      variant: {
        default: "",
        glow: "shadow-[0_4px_20px_rgba(34,211,238,0.25),inset_0_0_0_1px_rgba(34,211,238,0.3)]"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
var GlassTooltip = forwardRef(
  ({
    content,
    children,
    variant,
    side = "top",
    align = "center",
    sideOffset = 8,
    delayDuration = 200,
    contentClassName,
    showArrow = true,
    open,
    defaultOpen,
    onOpenChange,
    ...props
  }, ref) => {
    return /* @__PURE__ */ jsx(TooltipPrimitive.Provider, { delayDuration, children: /* @__PURE__ */ jsxs(
      TooltipPrimitive.Root,
      {
        open,
        defaultOpen,
        onOpenChange,
        ...props,
        children: [
          /* @__PURE__ */ jsx(TooltipPrimitive.Trigger, { asChild: true, children }),
          /* @__PURE__ */ jsx(TooltipPrimitive.Portal, { children: /* @__PURE__ */ jsxs(
            TooltipPrimitive.Content,
            {
              ref,
              side,
              align,
              sideOffset,
              className: cn(
                glassTooltipVariants({ variant }),
                contentClassName
              ),
              children: [
                content,
                showArrow && /* @__PURE__ */ jsx(
                  TooltipPrimitive.Arrow,
                  {
                    className: "fill-glass-card",
                    width: 12,
                    height: 6
                  }
                )
              ]
            }
          ) })
        ]
      }
    ) });
  }
);
GlassTooltip.displayName = "GlassTooltip";
var GlassTooltipProvider = TooltipPrimitive.Provider;
var glassTabsListVariants = cva(
  [
    "inline-flex items-center gap-1 p-1",
    "bg-glass-card/50 backdrop-blur-glass",
    "border border-white/12 rounded-xl"
  ],
  {
    variants: {
      variant: {
        default: "",
        pills: "bg-transparent border-0 p-0",
        underline: "bg-transparent border-0 border-b border-white/12 rounded-none p-0 pb-px"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
var glassTabsTriggerVariants = cva(
  [
    "relative px-4 py-2 rounded-lg",
    "text-sm font-medium text-white/60",
    "transition-all duration-200",
    "hover:text-white",
    "focus:outline-none focus-visible:ring-2 focus-visible:ring-glass-cyan focus-visible:ring-inset",
    "disabled:opacity-50 disabled:cursor-not-allowed"
  ],
  {
    variants: {
      variant: {
        default: [
          "data-[state=active]:text-white",
          "data-[state=active]:bg-white/10"
        ],
        pills: [
          "data-[state=active]:text-glass-dark",
          "data-[state=active]:bg-glass-cyan"
        ],
        underline: ["rounded-none", "data-[state=active]:text-white"]
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
var GlassTabsList = forwardRef(
  ({ className, variant, ...props }, ref) => /* @__PURE__ */ jsx(
    TabsPrimitive.List,
    {
      ref,
      className: cn(glassTabsListVariants({ variant }), className),
      ...props
    }
  )
);
GlassTabsList.displayName = "GlassTabsList";
var GlassTabsTrigger = forwardRef(({ className, variant, icon, children, ...props }, ref) => /* @__PURE__ */ jsxs(
  TabsPrimitive.Trigger,
  {
    ref,
    className: cn(glassTabsTriggerVariants({ variant }), className),
    ...props,
    children: [
      icon && /* @__PURE__ */ jsx("span", { className: "mr-2", children: icon }),
      children
    ]
  }
));
GlassTabsTrigger.displayName = "GlassTabsTrigger";
var GlassTabsContent = forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx(
  TabsPrimitive.Content,
  {
    ref,
    className: cn(
      "mt-4 focus:outline-none",
      "data-[state=active]:animate-fade-in",
      className
    ),
    ...props
  }
));
GlassTabsContent.displayName = "GlassTabsContent";
var GlassTabs = forwardRef(
  ({ className, children, ...props }, ref) => /* @__PURE__ */ jsx(
    TabsPrimitive.Root,
    {
      ref,
      className: cn("w-full", className),
      ...props,
      children
    }
  )
);
GlassTabs.displayName = "GlassTabs";
var glassToastVariants = cva(
  [
    "relative flex items-start gap-3 p-4 rounded-xl",
    "bg-glass-card/95 backdrop-blur-glass",
    "border border-white/12",
    "shadow-glass",
    "min-w-[300px] max-w-[400px]"
  ],
  {
    variants: {
      variant: {
        default: "",
        success: "border-emerald-500/30",
        error: "border-red-500/30",
        warning: "border-amber-500/30",
        info: "border-blue-500/30"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
var variantIcons = {
  default: /* @__PURE__ */ jsx(
    "svg",
    {
      className: "w-5 h-5 text-glass-cyan",
      fill: "none",
      stroke: "currentColor",
      viewBox: "0 0 24 24",
      children: /* @__PURE__ */ jsx(
        "path",
        {
          strokeLinecap: "round",
          strokeLinejoin: "round",
          strokeWidth: 2,
          d: "M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
        }
      )
    }
  ),
  success: /* @__PURE__ */ jsx(
    "svg",
    {
      className: "w-5 h-5 text-emerald-400",
      fill: "none",
      stroke: "currentColor",
      viewBox: "0 0 24 24",
      children: /* @__PURE__ */ jsx(
        "path",
        {
          strokeLinecap: "round",
          strokeLinejoin: "round",
          strokeWidth: 2,
          d: "M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        }
      )
    }
  ),
  error: /* @__PURE__ */ jsx(
    "svg",
    {
      className: "w-5 h-5 text-red-400",
      fill: "none",
      stroke: "currentColor",
      viewBox: "0 0 24 24",
      children: /* @__PURE__ */ jsx(
        "path",
        {
          strokeLinecap: "round",
          strokeLinejoin: "round",
          strokeWidth: 2,
          d: "M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z"
        }
      )
    }
  ),
  warning: /* @__PURE__ */ jsx(
    "svg",
    {
      className: "w-5 h-5 text-amber-400",
      fill: "none",
      stroke: "currentColor",
      viewBox: "0 0 24 24",
      children: /* @__PURE__ */ jsx(
        "path",
        {
          strokeLinecap: "round",
          strokeLinejoin: "round",
          strokeWidth: 2,
          d: "M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z"
        }
      )
    }
  ),
  info: /* @__PURE__ */ jsx(
    "svg",
    {
      className: "w-5 h-5 text-blue-400",
      fill: "none",
      stroke: "currentColor",
      viewBox: "0 0 24 24",
      children: /* @__PURE__ */ jsx(
        "path",
        {
          strokeLinecap: "round",
          strokeLinejoin: "round",
          strokeWidth: 2,
          d: "M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
        }
      )
    }
  )
};
var GlassToast = forwardRef(
  ({ title, description, variant = "default", action, onClose }, ref) => {
    return /* @__PURE__ */ jsxs(
      motion.div,
      {
        ref,
        layout: true,
        initial: { opacity: 0, y: -20, scale: 0.95 },
        animate: { opacity: 1, y: 0, scale: 1 },
        exit: { opacity: 0, y: -20, scale: 0.95 },
        transition: { duration: 0.2, ease: [0.4, 0, 0.2, 1] },
        className: cn(glassToastVariants({ variant })),
        role: "alert",
        "aria-live": "polite",
        children: [
          /* @__PURE__ */ jsx("div", { className: "shrink-0 mt-0.5", children: variantIcons[variant ?? "default"] }),
          /* @__PURE__ */ jsxs("div", { className: "flex-1 min-w-0", children: [
            title && /* @__PURE__ */ jsx("p", { className: "text-sm font-medium text-white", children: title }),
            description && /* @__PURE__ */ jsx("p", { className: "text-sm text-white/60 mt-0.5", children: description }),
            action && /* @__PURE__ */ jsx(
              "button",
              {
                onClick: action.onClick,
                className: "text-sm font-medium text-glass-cyan hover:text-glass-cyan/80 mt-2 transition-colors",
                children: action.label
              }
            )
          ] }),
          onClose && /* @__PURE__ */ jsx(
            "button",
            {
              onClick: onClose,
              className: "shrink-0 p-1 rounded-lg text-white/50 hover:text-white hover:bg-white/10 transition-colors",
              "aria-label": "Close notification",
              children: /* @__PURE__ */ jsx(
                "svg",
                {
                  className: "w-4 h-4",
                  fill: "none",
                  stroke: "currentColor",
                  viewBox: "0 0 24 24",
                  children: /* @__PURE__ */ jsx(
                    "path",
                    {
                      strokeLinecap: "round",
                      strokeLinejoin: "round",
                      strokeWidth: 2,
                      d: "M6 18L18 6M6 6l12 12"
                    }
                  )
                }
              )
            }
          )
        ]
      }
    );
  }
);
GlassToast.displayName = "GlassToast";
var ToastContext = createContext(void 0);
var useToastContext = () => {
  const context = useContext(ToastContext);
  if (!context) {
    throw new Error("useToastContext must be used within ToastProvider");
  }
  return context;
};
var positionClasses = {
  "top-right": "top-4 right-4",
  "top-left": "top-4 left-4",
  "bottom-right": "bottom-4 right-4",
  "bottom-left": "bottom-4 left-4",
  "top-center": "top-4 left-1/2 -translate-x-1/2",
  "bottom-center": "bottom-4 left-1/2 -translate-x-1/2"
};
var ToastProvider = ({
  children,
  position = "top-right",
  maxToasts = 5
}) => {
  const [toasts, setToasts] = useState([]);
  const addToast = useCallback(
    (toast) => {
      const id = `toast-${Date.now()}-${Math.random().toString(36).substr(2, 9)}`;
      const duration = toast.duration ?? 5e3;
      setToasts((prev) => {
        const newToasts = [...prev, { ...toast, id }];
        return newToasts.slice(-maxToasts);
      });
      if (duration > 0) {
        setTimeout(() => {
          setToasts((prev) => prev.filter((t) => t.id !== id));
        }, duration);
      }
      return id;
    },
    [maxToasts]
  );
  const removeToast = useCallback((id) => {
    setToasts((prev) => prev.filter((t) => t.id !== id));
  }, []);
  const clearToasts = useCallback(() => {
    setToasts([]);
  }, []);
  return /* @__PURE__ */ jsxs(
    ToastContext.Provider,
    {
      value: { toasts, addToast, removeToast, clearToasts },
      children: [
        children,
        /* @__PURE__ */ jsx(
          "div",
          {
            className: cn(
              "fixed z-[9999] flex flex-col gap-2",
              positionClasses[position]
            ),
            "aria-label": "Notifications",
            children: /* @__PURE__ */ jsx(AnimatePresence, { mode: "popLayout", children: toasts.map((toast) => /* @__PURE__ */ jsx(
              GlassToast,
              {
                ...toast,
                onClose: () => removeToast(toast.id)
              },
              toast.id
            )) })
          }
        )
      ]
    }
  );
};
var glassProgressVariants = cva(
  [
    "relative overflow-hidden",
    "bg-glass-card/50 backdrop-blur-glass",
    "border border-white/12"
  ],
  {
    variants: {
      size: {
        sm: "h-1.5 rounded-full",
        md: "h-2.5 rounded-full",
        lg: "h-4 rounded-full"
      },
      variant: {
        default: "",
        glow: "shadow-glass-glow"
      }
    },
    defaultVariants: {
      size: "md",
      variant: "default"
    }
  }
);
var glassProgressBarVariants = cva(
  "h-full rounded-full transition-all duration-300",
  {
    variants: {
      color: {
        cyan: "bg-glass-cyan",
        violet: "bg-glass-violet",
        pink: "bg-glass-pink",
        success: "bg-emerald-500",
        warning: "bg-amber-500",
        error: "bg-red-500",
        aurora: "aurora-gradient animate-aurora bg-[length:200%_200%]"
      }
    },
    defaultVariants: {
      color: "cyan"
    }
  }
);
var GlassProgress = forwardRef(
  ({
    value = 0,
    indeterminate = false,
    showLabel = false,
    labelPosition = "outside",
    formatLabel = (v) => `${Math.round(v)}%`,
    size,
    variant,
    color,
    className,
    ...props
  }, ref) => {
    const clampedValue = Math.min(100, Math.max(0, value));
    return /* @__PURE__ */ jsxs("div", { className: cn("w-full", className), children: [
      showLabel && labelPosition === "outside" && /* @__PURE__ */ jsxs("div", { className: "flex justify-between items-center mb-2", children: [
        /* @__PURE__ */ jsx("span", { className: "text-sm text-white/70", children: "Progress" }),
        /* @__PURE__ */ jsx("span", { className: "text-sm font-medium text-white", children: formatLabel(clampedValue) })
      ] }),
      /* @__PURE__ */ jsx(
        "div",
        {
          ref,
          role: "progressbar",
          "aria-valuenow": indeterminate ? void 0 : clampedValue,
          "aria-valuemin": 0,
          "aria-valuemax": 100,
          className: cn(glassProgressVariants({ size, variant })),
          ...props,
          children: indeterminate ? /* @__PURE__ */ jsx(
            motion.div,
            {
              className: cn(glassProgressBarVariants({ color }), "w-1/3"),
              animate: {
                x: ["-100%", "400%"]
              },
              transition: {
                duration: 1.5,
                repeat: Infinity,
                ease: "easeInOut"
              }
            }
          ) : /* @__PURE__ */ jsx(
            motion.div,
            {
              className: cn(glassProgressBarVariants({ color })),
              initial: { width: 0 },
              animate: { width: `${clampedValue}%` },
              transition: { duration: 0.3, ease: [0.4, 0, 0.2, 1] },
              children: showLabel && labelPosition === "inside" && size === "lg" && /* @__PURE__ */ jsx("span", { className: "absolute inset-0 flex items-center justify-center text-xs font-medium text-white", children: formatLabel(clampedValue) })
            }
          )
        }
      )
    ] });
  }
);
GlassProgress.displayName = "GlassProgress";
var colorClasses = {
  cyan: "stroke-glass-cyan",
  violet: "stroke-glass-violet",
  pink: "stroke-glass-pink",
  success: "stroke-emerald-500",
  warning: "stroke-amber-500",
  error: "stroke-red-500"
};
var GlassCircularProgress = forwardRef(
  ({
    value = 0,
    indeterminate = false,
    size = 48,
    strokeWidth = 4,
    color = "cyan",
    showLabel = false,
    label,
    className,
    ...props
  }, ref) => {
    const clampedValue = Math.min(100, Math.max(0, value));
    const radius = (size - strokeWidth) / 2;
    const circumference = radius * 2 * Math.PI;
    const offset = circumference - clampedValue / 100 * circumference;
    return /* @__PURE__ */ jsxs("div", { className: "relative inline-flex items-center justify-center", children: [
      /* @__PURE__ */ jsxs(
        "svg",
        {
          ref,
          width: size,
          height: size,
          viewBox: `0 0 ${size} ${size}`,
          className: cn("transform -rotate-90", className),
          ...props,
          children: [
            /* @__PURE__ */ jsx(
              "circle",
              {
                cx: size / 2,
                cy: size / 2,
                r: radius,
                fill: "none",
                stroke: "currentColor",
                strokeWidth,
                className: "text-white/10"
              }
            ),
            indeterminate ? /* @__PURE__ */ jsx(
              motion.circle,
              {
                cx: size / 2,
                cy: size / 2,
                r: radius,
                fill: "none",
                strokeWidth,
                strokeLinecap: "round",
                className: colorClasses[color],
                strokeDasharray: circumference,
                strokeDashoffset: circumference * 0.75,
                animate: { rotate: 360 },
                transition: {
                  duration: 1,
                  repeat: Infinity,
                  ease: "linear"
                },
                style: { transformOrigin: "center" }
              }
            ) : /* @__PURE__ */ jsx(
              motion.circle,
              {
                cx: size / 2,
                cy: size / 2,
                r: radius,
                fill: "none",
                strokeWidth,
                strokeLinecap: "round",
                className: colorClasses[color],
                strokeDasharray: circumference,
                initial: { strokeDashoffset: circumference },
                animate: { strokeDashoffset: offset },
                transition: { duration: 0.5, ease: [0.4, 0, 0.2, 1] }
              }
            )
          ]
        }
      ),
      (showLabel || label) && !indeterminate && /* @__PURE__ */ jsx("div", { className: "absolute inset-0 flex items-center justify-center", children: label || /* @__PURE__ */ jsxs("span", { className: "text-xs font-medium text-white", children: [
        Math.round(clampedValue),
        "%"
      ] }) })
    ] });
  }
);
GlassCircularProgress.displayName = "GlassCircularProgress";
var glassAvatarVariants = cva(
  [
    "relative inline-flex items-center justify-center",
    "bg-glass-card/70 backdrop-blur-glass",
    "border border-white/12",
    "overflow-hidden",
    "shrink-0"
  ],
  {
    variants: {
      size: {
        xs: "w-6 h-6 text-xs",
        sm: "w-8 h-8 text-sm",
        md: "w-10 h-10 text-base",
        lg: "w-12 h-12 text-lg",
        xl: "w-16 h-16 text-xl",
        "2xl": "w-20 h-20 text-2xl"
      },
      shape: {
        circle: "rounded-full",
        square: "rounded-xl"
      },
      variant: {
        default: "",
        gradient: "bg-gradient-to-br from-glass-cyan via-glass-violet to-glass-pink",
        glow: "shadow-glass-glow border-glass-cyan/30"
      }
    },
    defaultVariants: {
      size: "md",
      shape: "circle",
      variant: "default"
    }
  }
);
var statusVariants = cva("absolute border-2 border-glass-dark rounded-full", {
  variants: {
    status: {
      online: "bg-emerald-500",
      offline: "bg-slate-500",
      away: "bg-amber-500",
      busy: "bg-red-500"
    },
    size: {
      xs: "w-1.5 h-1.5 bottom-0 right-0",
      sm: "w-2 h-2 bottom-0 right-0",
      md: "w-2.5 h-2.5 bottom-0 right-0",
      lg: "w-3 h-3 bottom-0.5 right-0.5",
      xl: "w-3.5 h-3.5 bottom-1 right-1",
      "2xl": "w-4 h-4 bottom-1 right-1"
    }
  },
  defaultVariants: {
    status: "offline",
    size: "md"
  }
});
var getInitials = (name) => {
  return name.split(" ").map((part) => part[0]).join("").toUpperCase().slice(0, 2);
};
var DefaultFallbackIcon = () => /* @__PURE__ */ jsx(
  "svg",
  {
    className: "w-1/2 h-1/2 text-white/60",
    fill: "currentColor",
    viewBox: "0 0 24 24",
    children: /* @__PURE__ */ jsx("path", { d: "M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z" })
  }
);
var GlassAvatar = forwardRef(
  ({
    src,
    alt,
    fallback,
    size,
    shape,
    variant,
    status,
    fallbackIcon,
    className,
    ...props
  }, ref) => {
    const [imageError, setImageError] = useState(false);
    const showFallback = !src || imageError;
    return /* @__PURE__ */ jsxs(
      "div",
      {
        ref,
        className: cn(glassAvatarVariants({ size, shape, variant }), className),
        children: [
          showFallback ? /* @__PURE__ */ jsx("span", { className: "font-medium text-white select-none", children: fallback ? getInitials(fallback) : fallbackIcon ? fallbackIcon : /* @__PURE__ */ jsx(DefaultFallbackIcon, {}) }) : /* @__PURE__ */ jsx(
            "img",
            {
              src,
              alt: alt || fallback || "Avatar",
              onError: () => setImageError(true),
              className: "w-full h-full object-cover",
              ...props
            }
          ),
          status && /* @__PURE__ */ jsx(
            "span",
            {
              className: cn(statusVariants({ status, size })),
              "aria-label": `Status: ${status}`
            }
          )
        ]
      }
    );
  }
);
GlassAvatar.displayName = "GlassAvatar";
var GlassAvatarGroup = forwardRef(
  ({ max, size = "md", children, className, ...props }, ref) => {
    const childArray = React4.Children.toArray(children);
    const excess = max ? Math.max(0, childArray.length - max) : 0;
    const visibleChildren = max ? childArray.slice(0, max) : childArray;
    const spacingClasses = {
      xs: "-space-x-2",
      sm: "-space-x-2.5",
      md: "-space-x-3",
      lg: "-space-x-4",
      xl: "-space-x-5",
      "2xl": "-space-x-6"
    };
    return /* @__PURE__ */ jsxs(
      "div",
      {
        ref,
        className: cn("flex items-center", spacingClasses[size], className),
        ...props,
        children: [
          visibleChildren.map((child, index) => /* @__PURE__ */ jsx(
            "div",
            {
              className: "relative ring-2 ring-glass-dark rounded-full",
              style: { zIndex: visibleChildren.length - index },
              children: React4.cloneElement(child, { size })
            },
            index
          )),
          excess > 0 && /* @__PURE__ */ jsx(
            "div",
            {
              className: "relative ring-2 ring-glass-dark rounded-full",
              style: { zIndex: 0 },
              children: /* @__PURE__ */ jsx(
                GlassAvatar,
                {
                  size,
                  fallback: `+${excess}`,
                  variant: "default"
                }
              )
            }
          )
        ]
      }
    );
  }
);
GlassAvatarGroup.displayName = "GlassAvatarGroup";
var glassBadgeVariants = cva(
  [
    "inline-flex items-center justify-center gap-1",
    "font-medium",
    "transition-all duration-200"
  ],
  {
    variants: {
      variant: {
        default: [
          "bg-glass-card/70 backdrop-blur-glass",
          "border border-white/12",
          "text-white"
        ],
        primary: [
          "bg-glass-cyan/20 backdrop-blur-glass",
          "border border-glass-cyan/30",
          "text-glass-cyan"
        ],
        secondary: [
          "bg-glass-violet/20 backdrop-blur-glass",
          "border border-glass-violet/30",
          "text-glass-violet"
        ],
        accent: [
          "bg-glass-pink/20 backdrop-blur-glass",
          "border border-glass-pink/30",
          "text-glass-pink"
        ],
        success: [
          "bg-emerald-500/20 backdrop-blur-glass",
          "border border-emerald-500/30",
          "text-emerald-400"
        ],
        warning: [
          "bg-amber-500/20 backdrop-blur-glass",
          "border border-amber-500/30",
          "text-amber-400"
        ],
        error: [
          "bg-red-500/20 backdrop-blur-glass",
          "border border-red-500/30",
          "text-red-400"
        ],
        outline: ["bg-transparent", "border border-white/20", "text-white"]
      },
      size: {
        sm: "h-5 px-2 text-xs rounded-md",
        md: "h-6 px-2.5 text-xs rounded-lg",
        lg: "h-7 px-3 text-sm rounded-lg"
      }
    },
    defaultVariants: {
      variant: "default",
      size: "md"
    }
  }
);
var dotColors = {
  cyan: "bg-glass-cyan",
  violet: "bg-glass-violet",
  pink: "bg-glass-pink",
  success: "bg-emerald-500",
  warning: "bg-amber-500",
  error: "bg-red-500"
};
var GlassBadge = forwardRef(
  ({
    children,
    variant,
    size,
    leftIcon,
    rightIcon,
    dot,
    dotColor = "cyan",
    onRemove,
    className,
    ...props
  }, ref) => {
    return /* @__PURE__ */ jsxs(
      "span",
      {
        ref,
        className: cn(glassBadgeVariants({ variant, size }), className),
        ...props,
        children: [
          dot && /* @__PURE__ */ jsx(
            "span",
            {
              className: cn("w-1.5 h-1.5 rounded-full", dotColors[dotColor]),
              "aria-hidden": "true"
            }
          ),
          leftIcon && /* @__PURE__ */ jsx("span", { className: "shrink-0", children: leftIcon }),
          children,
          rightIcon && /* @__PURE__ */ jsx("span", { className: "shrink-0", children: rightIcon }),
          onRemove && /* @__PURE__ */ jsx(
            "button",
            {
              onClick: onRemove,
              className: "shrink-0 ml-0.5 p-0.5 rounded hover:bg-white/10 transition-colors",
              "aria-label": "Remove",
              children: /* @__PURE__ */ jsx(
                "svg",
                {
                  className: "w-3 h-3",
                  fill: "none",
                  stroke: "currentColor",
                  viewBox: "0 0 24 24",
                  children: /* @__PURE__ */ jsx(
                    "path",
                    {
                      strokeLinecap: "round",
                      strokeLinejoin: "round",
                      strokeWidth: 2,
                      d: "M6 18L18 6M6 6l12 12"
                    }
                  )
                }
              )
            }
          )
        ]
      }
    );
  }
);
GlassBadge.displayName = "GlassBadge";
var SearchIcon = () => /* @__PURE__ */ jsx("svg", { className: "w-5 h-5 text-white/50", fill: "none", stroke: "currentColor", viewBox: "0 0 24 24", children: /* @__PURE__ */ jsx("path", { strokeLinecap: "round", strokeLinejoin: "round", strokeWidth: 2, d: "M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z" }) });
var overlayVariants2 = {
  hidden: { opacity: 0 },
  visible: { opacity: 1 }
};
var contentVariants3 = {
  hidden: { opacity: 0, scale: 0.95, y: -20 },
  visible: { opacity: 1, scale: 1, y: 0 }
};
var GlassCommandPalette = forwardRef(
  ({
    commands,
    open: controlledOpen,
    onOpenChange,
    placeholder = "Search commands...",
    emptyMessage = "No commands found.",
    filter: customFilter
  }, ref) => {
    const [internalOpen, setInternalOpen] = useState(false);
    const [search, setSearch] = useState("");
    const [selectedIndex, setSelectedIndex] = useState(0);
    const listRef = useRef(null);
    const inputRef = useRef(null);
    const open = controlledOpen ?? internalOpen;
    const setOpen = onOpenChange ?? setInternalOpen;
    const defaultFilter = (command, searchTerm) => {
      const normalizedSearch = searchTerm.toLowerCase();
      return command.label.toLowerCase().includes(normalizedSearch) || command.description?.toLowerCase().includes(normalizedSearch) || command.group?.toLowerCase().includes(normalizedSearch);
    };
    const filterFn = customFilter ?? defaultFilter;
    const filteredCommands = search ? commands.filter((cmd) => filterFn(cmd, search)) : commands;
    const groupedCommands = filteredCommands.reduce((acc, command) => {
      const group = command.group || "Commands";
      if (!acc[group]) {
        acc[group] = [];
      }
      acc[group].push(command);
      return acc;
    }, {});
    const flatCommands = Object.values(groupedCommands).flat();
    useEffect(() => {
      const handleKeyDown2 = (e) => {
        if ((e.metaKey || e.ctrlKey) && e.key === "k") {
          e.preventDefault();
          setOpen(!open);
        }
      };
      document.addEventListener("keydown", handleKeyDown2);
      return () => document.removeEventListener("keydown", handleKeyDown2);
    }, [open, setOpen]);
    useEffect(() => {
      if (open) {
        setSearch("");
        setSelectedIndex(0);
        setTimeout(() => inputRef.current?.focus(), 0);
      }
    }, [open]);
    const handleKeyDown = useCallback(
      (e) => {
        switch (e.key) {
          case "ArrowDown":
            e.preventDefault();
            setSelectedIndex((i) => Math.min(i + 1, flatCommands.length - 1));
            break;
          case "ArrowUp":
            e.preventDefault();
            setSelectedIndex((i) => Math.max(i - 1, 0));
            break;
          case "Enter":
            e.preventDefault();
            const command = flatCommands[selectedIndex];
            if (command && !command.disabled) {
              command.onSelect();
              setOpen(false);
            }
            break;
          case "Escape":
            setOpen(false);
            break;
        }
      },
      [flatCommands, selectedIndex, setOpen]
    );
    useEffect(() => {
      const selectedElement = listRef.current?.querySelector(
        `[data-index="${selectedIndex}"]`
      );
      selectedElement?.scrollIntoView({ block: "nearest" });
    }, [selectedIndex]);
    useEffect(() => {
      setSelectedIndex(0);
    }, [search]);
    return /* @__PURE__ */ jsx(DialogPrimitive.Root, { open, onOpenChange: setOpen, children: /* @__PURE__ */ jsx(AnimatePresence, { children: open && /* @__PURE__ */ jsxs(DialogPrimitive.Portal, { forceMount: true, children: [
      /* @__PURE__ */ jsx(DialogPrimitive.Overlay, { asChild: true, children: /* @__PURE__ */ jsx(
        motion.div,
        {
          className: "fixed inset-0 bg-black/60 backdrop-blur-sm z-[100]",
          variants: overlayVariants2,
          initial: "hidden",
          animate: "visible",
          exit: "hidden",
          transition: { duration: 0.2 }
        }
      ) }),
      /* @__PURE__ */ jsx(DialogPrimitive.Content, { asChild: true, children: /* @__PURE__ */ jsxs(
        motion.div,
        {
          ref,
          className: cn(
            "fixed left-1/2 top-[20%] -translate-x-1/2",
            "w-full max-w-xl",
            "bg-glass-card/95 backdrop-blur-glass-lg",
            "border border-white/12 rounded-2xl",
            "shadow-glass-lg",
            "overflow-hidden",
            "z-[100]",
            "focus:outline-none"
          ),
          variants: contentVariants3,
          initial: "hidden",
          animate: "visible",
          exit: "hidden",
          transition: { duration: 0.2, ease: [0.4, 0, 0.2, 1] },
          onKeyDown: handleKeyDown,
          children: [
            /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-3 px-4 py-3 border-b border-white/12", children: [
              /* @__PURE__ */ jsx(SearchIcon, {}),
              /* @__PURE__ */ jsx(
                "input",
                {
                  ref: inputRef,
                  type: "text",
                  value: search,
                  onChange: (e) => setSearch(e.target.value),
                  placeholder,
                  className: cn(
                    "flex-1 bg-transparent",
                    "text-white placeholder:text-white/40",
                    "focus:outline-none"
                  ),
                  autoComplete: "off",
                  autoCorrect: "off",
                  spellCheck: "false"
                }
              ),
              /* @__PURE__ */ jsx("kbd", { className: "hidden sm:inline-flex items-center gap-1 px-2 py-1 text-xs text-white/40 bg-white/5 rounded border border-white/10", children: "ESC" })
            ] }),
            /* @__PURE__ */ jsx(
              "div",
              {
                ref: listRef,
                className: "max-h-[300px] overflow-y-auto p-2",
                role: "listbox",
                children: flatCommands.length === 0 ? /* @__PURE__ */ jsx("div", { className: "py-8 text-center text-white/50", children: emptyMessage }) : Object.entries(groupedCommands).map(([group, items]) => /* @__PURE__ */ jsxs("div", { className: "mb-2 last:mb-0", children: [
                  /* @__PURE__ */ jsx("div", { className: "px-3 py-2 text-xs font-medium text-white/40 uppercase", children: group }),
                  items.map((command) => {
                    const index = flatCommands.indexOf(command);
                    const isSelected = index === selectedIndex;
                    return /* @__PURE__ */ jsxs(
                      "button",
                      {
                        "data-index": index,
                        onClick: () => {
                          if (!command.disabled) {
                            command.onSelect();
                            setOpen(false);
                          }
                        },
                        onMouseEnter: () => setSelectedIndex(index),
                        disabled: command.disabled,
                        className: cn(
                          "w-full flex items-center gap-3 px-3 py-2.5 rounded-xl",
                          "text-left transition-colors duration-100",
                          isSelected ? "bg-white/10 text-white" : "text-white/70 hover:text-white",
                          command.disabled && "opacity-50 cursor-not-allowed"
                        ),
                        role: "option",
                        "aria-selected": isSelected,
                        children: [
                          command.icon && /* @__PURE__ */ jsx("span", { className: "shrink-0 w-5 h-5", children: command.icon }),
                          /* @__PURE__ */ jsxs("div", { className: "flex-1 min-w-0", children: [
                            /* @__PURE__ */ jsx("div", { className: "font-medium truncate", children: command.label }),
                            command.description && /* @__PURE__ */ jsx("div", { className: "text-sm text-white/50 truncate", children: command.description })
                          ] }),
                          command.shortcut && /* @__PURE__ */ jsx("kbd", { className: "shrink-0 px-2 py-1 text-xs text-white/40 bg-white/5 rounded border border-white/10", children: command.shortcut })
                        ]
                      },
                      command.id
                    );
                  })
                ] }, group))
              }
            ),
            /* @__PURE__ */ jsxs("div", { className: "flex items-center justify-between px-4 py-2 border-t border-white/12 text-xs text-white/40", children: [
              /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-4", children: [
                /* @__PURE__ */ jsxs("span", { className: "flex items-center gap-1", children: [
                  /* @__PURE__ */ jsx("kbd", { className: "px-1.5 py-0.5 bg-white/5 rounded border border-white/10", children: "\u2191\u2193" }),
                  /* @__PURE__ */ jsx("span", { children: "Navigate" })
                ] }),
                /* @__PURE__ */ jsxs("span", { className: "flex items-center gap-1", children: [
                  /* @__PURE__ */ jsx("kbd", { className: "px-1.5 py-0.5 bg-white/5 rounded border border-white/10", children: "\u21B5" }),
                  /* @__PURE__ */ jsx("span", { children: "Select" })
                ] })
              ] }),
              /* @__PURE__ */ jsxs("span", { className: "flex items-center gap-1", children: [
                /* @__PURE__ */ jsx("kbd", { className: "px-1.5 py-0.5 bg-white/5 rounded border border-white/10", children: "\u2318K" }),
                /* @__PURE__ */ jsx("span", { children: "Toggle" })
              ] })
            ] })
          ]
        }
      ) })
    ] }) }) });
  }
);
GlassCommandPalette.displayName = "GlassCommandPalette";

// src/hooks/useToast.ts
function useToast() {
  const { toasts, addToast, removeToast, clearToasts } = useToastContext();
  const toast = (options) => {
    return addToast(options);
  };
  const success = (title, description) => {
    return addToast({ variant: "success", title, description });
  };
  const error = (title, description) => {
    return addToast({ variant: "error", title, description });
  };
  const warning = (title, description) => {
    return addToast({ variant: "warning", title, description });
  };
  const info = (title, description) => {
    return addToast({ variant: "info", title, description });
  };
  return {
    toast,
    success,
    error,
    warning,
    info,
    dismiss: removeToast,
    dismissAll: clearToasts,
    toasts
  };
}
function useModal(options = {}) {
  const { defaultOpen = false, onOpen, onClose } = options;
  const [isOpen, setIsOpen] = useState(defaultOpen);
  const open = useCallback(() => {
    setIsOpen(true);
    onOpen?.();
  }, [onOpen]);
  const close = useCallback(() => {
    setIsOpen(false);
    onClose?.();
  }, [onClose]);
  const toggle = useCallback(() => {
    setIsOpen((prev) => {
      const next = !prev;
      if (next) {
        onOpen?.();
      } else {
        onClose?.();
      }
      return next;
    });
  }, [onOpen, onClose]);
  const setOpen = useCallback(
    (open2) => {
      setIsOpen(open2);
      if (open2) {
        onOpen?.();
      } else {
        onClose?.();
      }
    },
    [onOpen, onClose]
  );
  return {
    isOpen,
    open,
    close,
    toggle,
    setOpen
  };
}
var STORAGE_KEY = "glass-ui-theme";
var ATTRIBUTE = "class";
function useTheme(options = {}) {
  const {
    defaultTheme = "system",
    storageKey = STORAGE_KEY,
    attribute = ATTRIBUTE
  } = options;
  const getInitialTheme = () => {
    if (typeof window === "undefined") {
      return defaultTheme;
    }
    const stored = localStorage.getItem(storageKey);
    if (stored && ["light", "dark", "system"].includes(stored)) {
      return stored;
    }
    return defaultTheme;
  };
  const [theme, setThemeState] = useState(getInitialTheme);
  const getSystemTheme = useCallback(() => {
    if (typeof window === "undefined") {
      return "dark";
    }
    return window.matchMedia("(prefers-color-scheme: dark)").matches ? "dark" : "light";
  }, []);
  const resolvedTheme = useMemo(() => {
    if (theme === "system") {
      return getSystemTheme();
    }
    return theme;
  }, [theme, getSystemTheme]);
  useEffect(() => {
    const root = document.documentElement;
    if (attribute === "class") {
      root.classList.remove("light", "dark");
      root.classList.add(resolvedTheme);
    } else {
      root.setAttribute(attribute, resolvedTheme);
    }
  }, [resolvedTheme, attribute]);
  useEffect(() => {
    if (typeof window === "undefined") return;
    const mediaQuery = window.matchMedia("(prefers-color-scheme: dark)");
    const handleChange = () => {
      if (theme === "system") {
        setThemeState("system");
      }
    };
    mediaQuery.addEventListener("change", handleChange);
    return () => mediaQuery.removeEventListener("change", handleChange);
  }, [theme]);
  const setTheme = useCallback(
    (newTheme) => {
      setThemeState(newTheme);
      if (typeof window !== "undefined") {
        localStorage.setItem(storageKey, newTheme);
      }
    },
    [storageKey]
  );
  const toggle = useCallback(() => {
    const newTheme = resolvedTheme === "dark" ? "light" : "dark";
    setTheme(newTheme);
  }, [resolvedTheme, setTheme]);
  return {
    theme,
    resolvedTheme,
    setTheme,
    toggle,
    isDark: resolvedTheme === "dark",
    isLight: resolvedTheme === "light"
  };
}
/**
 * @mawtech/glass-ui
 * Beautiful Apple-style glassmorphism React components
 *
 * @package @mawtech/glass-ui
 * @author MAW Tech Solutions
 * @see https://mawtechsolutions.com
 * @license MIT
 */

export { GlassAvatar, GlassAvatarGroup, GlassBadge, GlassButton, GlassCard, GlassCheckbox, GlassCircularProgress, GlassCommandPalette, GlassDropdown, GlassDropdownItem, GlassDropdownLabel, GlassDropdownSeparator, GlassInput, GlassModal, GlassModalClose, GlassModalTrigger, GlassNavItem, GlassNavbar, GlassProgress, GlassSelect, GlassSidebar, GlassSidebarItem, GlassSwitch, GlassTabs, GlassTabsContent, GlassTabsList, GlassTabsTrigger, GlassTextarea, GlassToast, GlassTooltip, GlassTooltipProvider, ToastProvider, cn, useModal, useTheme, useToast, useToastContext };
